ethcore/src/miner/miner.rs

```diff
@@ -1114,77 +1115,26 @@ impl miner::MinerService for Miner {
 					max_len,
 					ordering,
 				},
-			)
+			);
+
+			pending.retain(|tx| {
+				filter.as_ref().map_or(true, |filter| {
+					filter.matches(tx.signed())
+				})
+			});
+
+			pending
```

Comment:
Just remove the semi-column and the `pending` here to let implicit returns do its thing?

Reply:
[Vec::retain](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain) doesn't return anything i.e, takes `&mut self`